### PR TITLE
Add line directives to preprocessed files

### PR DESCRIPTION
**Pull request by lefessan**
These directives should improve localisation of errors.

### CODE REVIEW

File: stdlib/sys.mlp [hunk 1]
```diff
@@ -17,10 +18,10 @@
    your changes will be lost.
 *)
 
-type backend_type = 
+type backend_type =
   | Native
   | Bytecode
-  | Other of string 
+  | Other of string
```
Comment (mshinwell) [inline stdlib/sys.mlp:24]: This appears to be a whitespace-only change

**Comment by mshinwell** [comment]
This seems like a good idea.

**Comment by alainfrisch** [comment]
The intel backend already have these line directives, and I confirm they are very useful.

What about the other backend-dependent files which are copied by the build system?  Applying the same technique would avoid editing by error the copies instead of the originals.

**Comment by lefessan** [comment]
@alainfrisch 
Indeed, we could do the same for copied files. I propose to merge this PR, and keep in mind the other additions to do them when we have time.

File: stdlib/sys.mlp (hunk 1 above)
Comment (lefessan) [inline stdlib/sys.mlp:24]: Yes, but removing whitespaces at end of line is always good, isn't it ?

**Comment by mshinwell** [comment]
@alainfrisch : Those other files are symlinked, not copied, right?

**Comment by lefessan** [comment]
They are copied on Windows.

**Comment by shindere** [comment]
Fabrice Le Fessant (2016/06/29 02:57 -0700):

> I propose to merge this PR, and keep in mind the other additions to do
> them when we have time.

Do we have an alternative place to "mind" to keep track of such things
that should be done?

File: stdlib/sys.mlp (hunk 1 above)
Comment (shindere) [inline stdlib/sys.mlp:24]: Fabrice Le Fessant (2016/06/29 02:58 -0700):

> Yes, but removing whitespaces at end of line is always good, isn't it
> ?

Per se it is, definitely. One may prefere to do it in distinct commits,
though.

**Comment by mshinwell** [comment]
@shindere I have started using the Jane Street style "CR" markings in comments in the code.  These have a few advantages: (a) persistent; (b) next to the code; (c) easy to grep for.  However I think this practice is neither "officially" sanctioned nor widespread.

**Comment by alainfrisch** [comment]
> Those other files are symlinked, not copied, right?

Even when they are (i.e. not under Windows), the filename in error message is still not the original path (I guess).

**Comment by shindere** [comment]
Okay @mshinwell, thanks. Could you please elaborate a bit on what these
CR markings are or perhaps provide a link? Thanks a lot in advance!

File: stdlib/sys.mlp (hunk 1 above)
Comment (damiendoligez) [inline stdlib/sys.mlp:24]: Removing whitespace adds noise to the PR. I usually do it as a big commit, using tools/check-typo to make sure I didn't miss any.

So: don't do it unless you're already changing the line for some other reason.

**Comment by damiendoligez** [comment]
> Even when they are (i.e. not under Windows), the filename in error message is still not the original path (I guess).

But emacs will correctly open the original rather than the link.

**Comment by alainfrisch** [comment]
Fair enough.  This is mostly for people developing under Windows, indeed.

**Pull request merged**